ballerina/tests/tests.bal

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/oauth2;
+import ballerina/http;
+
+configurable string clientId = ?;
+configurable string clientSecret = ?;
+configurable string refreshToken = ?;
+
+OAuth2RefreshTokenGrantConfig auth = {
+       clientId: clientId,
+       clientSecret: clientSecret,
+       refreshToken: refreshToken,
+       credentialBearer: oauth2:POST_BODY_BEARER
+};
+
+ConnectionConfig config = {auth:auth};
+final Client baseClient = check new Client(config);
+
+//Create a new Cart
+@test:Config {}
+isolated function testCreateCart() {
+
+    SimplePublicObjectInputForCreate payload = {
+      "properties": {
+        "hs_cart_name": "Optic",
+        "hs_external_cart_id": "1234567890",
+        "hs_external_status": "pending",
+        "hs_source_store": "XYZ - USA",
+        "hs_total_price": "500",
+        "hs_currency_code": "USD",
+        "hs_cart_discount": "12",
+        "hs_tax": "36.25",
+        "hs_shipping_cost": "0",
+        "hs_tags": "frames, lenses"
+      }
+    };
+
+    SimplePublicObject|error response =  baseClient ->/carts.post(payload = payload);
```

Comment:
Please use a formatter to format the code. Ballerina VS Code plugin can format the code, or you can use the `bal format` command to format a Ballerina file. 

```suggestion
    SimplePublicObject|error response = baseClient ->/carts.post(payload = payload);
```